hypertrace-metrics-exporter/hypertrace-metrics-exporter/src/main/java/org/hypertrace/metrics/exporter/consumer/MetricsKafkaConsumer.java

```diff
@@ -0,0 +1,115 @@
+import org.slf4j.LoggerFactory;
+
+public class MetricsKafkaConsumer implements Runnable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetricsKafkaConsumer.class);
+
+  private static final int CONSUMER_POLL_TIMEOUT_MS = 100;
+  private static final int QUEUE_WAIT_TIME_MS = 500;
+  private static final int WAIT_TIME_MS = 100;
+
+  private static final String KAFKA_CONFIG_KEY = "kafka.config";
+  private static final String INPUT_TOPIC_KEY = "input.topic";
+
+  private final KafkaConsumer<byte[], byte[]> consumer;
+  private final InMemoryMetricsProducer inMemoryMetricsProducer;
+
+  public MetricsKafkaConsumer(Config config, InMemoryMetricsProducer inMemoryMetricsProducer) {
+    consumer = new KafkaConsumer<>(prepareProperties(config.getConfig(KAFKA_CONFIG_KEY)));
+    consumer.subscribe(Collections.singletonList(config.getString(INPUT_TOPIC_KEY)));
+    this.inMemoryMetricsProducer = inMemoryMetricsProducer;
+  }
+
+  public void run() {
+    while (true && !Thread.currentThread().isInterrupted()) {
+      List<ResourceMetrics> resourceMetrics = new ArrayList<>();
+
+      ConsumerRecords<byte[], byte[]> records =
+          consumer.poll(Duration.ofMillis(CONSUMER_POLL_TIMEOUT_MS));
+
+      records.forEach(
+          record -> {
+            try {
+              resourceMetrics.add(ResourceMetrics.parseFrom(record.value()));
+            } catch (InvalidProtocolBufferException e) {
+              LOGGER.warn("Skipping record due to error", e);
+            }
+          });
+
+      resourceMetrics.forEach(
+          rm -> {
+            try {
+              List<MetricData> metricData = OtlpProtoToMetricDataConverter.toMetricData(rm);
+              boolean result = false;
+              while (!result) {
+                result = inMemoryMetricsProducer.addMetricData(metricData);
+                waitForSec(QUEUE_WAIT_TIME_MS);
+              }
+            } catch (Exception e) {
+              LOGGER.debug("skipping the resource metrics due to error: {}", e);
+            }
+          });
+
+      waitForSec(WAIT_TIME_MS);
+    }
+  }
+
+  public void close() {
+    consumer.close();
+  }
+
+  private Properties prepareProperties(Config config) {
+    Map<String, Object> overrideProperties = new HashMap();
+    config.entrySet().stream()
+        .forEach(e -> overrideProperties.put(e.getKey(), config.getString(e.getKey())));
+
+    Map<String, Object> baseProperties = getBaseProperties();
+    overrideProperties.forEach(baseProperties::put);
+
+    Properties properties = new Properties();
+    properties.putAll(baseProperties);
+    return properties;
+  }
+
+  private Map<String, Object> getBaseProperties() {
+    Map<String, Object> baseProperties = new HashMap<>();
+    baseProperties.put(ConsumerConfig.GROUP_ID_CONFIG, "hypertrace-metrics-exporter");
+    baseProperties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+    baseProperties.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
```

Comment:
As mentioned in the description that we will handle the manual coming in the second iteration.